### ISSUE TITLE
Checkpointer should not use the unstable Serialization package.

### ISSUE THREAD

**Issue by ali-ramadhan** [ali-ramadhan/Oceananigans.jl#30]
Should switch to something more stable/persistent like [HDF5.jl](https://github.com/JuliaIO/HDF5.jl) or [JLD.jl](https://github.com/JuliaIO/JLD.jl).

**Comment by ali-ramadhan** [comment]
Also check out [JLD2.jl](https://github.com/JuliaIO/JLD2.jl).

**Comment by ali-ramadhan** [comment]
Should do JLD2.jl but maybe don't serialize the FFTW plans. It does not play well with the Serialization package.

**Comment by glwagner** [comment]
Perhaps it makes sense to design a Checkpoint type that contains the minimal info needed to reinstantiate a model, and save that?

It’s possible to run into issues when functions are attributes of the model (for example, user-defined forcing functions). I think serialization *can* solve that problem, but is undesirable for other cases. 

Possibly, this shortcoming can be addressed by a contribution to JLD2 itself.

**Comment by ali-ramadhan** [comment]
I just did some refactoring so that the Poisson solver parameters (including any FFTW or CuFFT plans) are not serialized to disk. They are rebuilt when restoring from a checkpoint.

I decided to go with [JLD.jl](https://github.com/JuliaIO/JLD.jl) as [JLD2.jl](https://github.com/JuliaIO/JLD2.jl) looks to be unmaintained since Julia 1.0= came out, which is a shame as it fits our needs better. But either way, checkpointing is done every 10k time steps so not too worried about speed right now.
